x-pack/plugins/threat_intelligence/public/containers/integrations_guard/integrations_guard.tsx

```diff
@@ -7,26 +7,78 @@
 
 import { EuiLoadingLogo } from '@elastic/eui';
 import React, { FC } from 'react';
+import { useQuery } from '@tanstack/react-query';
 import { EmptyPage } from '../../modules/empty_page';
 import { useIndicatorsTotalCount } from '../../modules/indicators';
 import { SecuritySolutionPluginTemplateWrapper } from '../security_solution_plugin_template_wrapper';
+import { useKibana } from '../../hooks';
+
+type IntegrationInstallStatus = 'installed' | 'installing' | 'install_failed';
+
+const INTEGRATIONS_URL = '/api/fleet/epm/packages';
+
+const INTEGRATIONS_STALE_TIME = 2000;
+
+export const THREAT_INTELLIGENCE_CATEGORY = 'threat_intel';
+
+export const THREAT_INTELLIGENCE_UTILITIES = 'ti_util';
+
+export const INSTALLATION_STATUS = {
+  Installed: 'installed',
+  Installing: 'installing',
+  InstallFailed: 'install_failed',
+  NotInstalled: 'not_installed',
+};
+
+interface IntegrationResponse {
+  items: Integration[];
+}
+
+interface Integration {
+  categories: string[];
+  id: string;
+  status: IntegrationInstallStatus;
+}
 
 /**
- * Renders children only if TI integrations are enabled
+ * Renders children only if TI integrations are enabled and indicators are received
  */
 export const IntegrationsGuard: FC = ({ children }) => {
-  const { count: indicatorsTotalCount, isLoading: isIndicatorsTotalCountLoading } =
+  const { http } = useKibana().services;
+
+  const { isLoading: indicatorsTotalCountLoading, count: indicatorsTotalCount } =
     useIndicatorsTotalCount();
 
-  if (isIndicatorsTotalCountLoading) {
+  // we only select integrations: (see https://github.com/elastic/security-team/issues/4374)
+  // - of status `installed`
+  // - with `threat_intel` category
+  // - excluding `ti_util` integration
+  const { isLoading: integrationLoading, data: installedTIIntegrations } = useQuery(
+    ['integrations'],
+    () => http.get<IntegrationResponse>(INTEGRATIONS_URL),
+    {
+      select: (data: IntegrationResponse) =>
+        data?.items.filter(
+          (pkg: any) =>
+            pkg.status === INSTALLATION_STATUS.Installed &&
+            pkg.categories.find((category: string) => category === THREAT_INTELLIGENCE_CATEGORY) !=
+              null &&
+            pkg.id !== THREAT_INTELLIGENCE_UTILITIES
+        ) || [],
+      staleTime: INTEGRATIONS_STALE_TIME,
```

Comment:
I believe `staleTime` doesn't solve the issue with the fleet endpoint 
```
Similarly, staleTime determines for how long a certain response will still be considered fresh (or not stale), dismissing the need for a new request.
```
from https://medium.com/doctolib/react-query-cachetime-vs-staletime-ec74defc483e

What I meant is some small timeout for the `http.get` itself. I don't know what the current default is but usually it's quite long, eg. 30s. So setting it to smth like 1 second will make sure whatever happens with the fleet request, our page will show up after one second at worst.

Reply:
yup fixed!